### ISSUE TITLE
mutinode ray error

### ISSUE THREAD

**Issue by ShoutaoGuo** [OpenRLHF/OpenRLHF#845]
When I use ray to conduct multinode training, I usually encounter the following error?

(ActorModelRayActor pid=11099, ip=172.24.168.115) [2025-03-09 13:44:58,420 E 11099 11160] core_worker.h:1564: Mismatched WorkerID: ignoring RPC for previous worker 0761da2bc149dd351cca0f0b53571a946891db77ca0ec340b4233089, current worker ID: 7373fe30e495eee8dcea24200851874fedaf6fa57508d89d98b97aa1

The following is my requirements:

accelerate
bitsandbytes
datasets
deepspeed==0.16.3
einops
isort
jsonlines
loralib
optimum
packaging
peft
pynvml>=12.0.0
ray[default]==2.42.0
tensorboard
vllm
torch
torchmetrics
tqdm
transformers==4.48.3
transformers_stream_generator
wandb
wheel

**Comment by ShoutaoGuo** [comment]
Besides, when I run the following code on single node, it can run without any error. But when I run the code on multi-node, it will encounter the above error and stay there.

python3 -m openrlhf.cli.train_ppo_ray \
  --ref_num_nodes 1 \
  --ref_num_gpus_per_node 2 \
  --reward_num_nodes 1 \
  --reward_num_gpus_per_node 2 \
  --critic_num_nodes 1 \
  --critic_num_gpus_per_node 2 \
  --actor_num_nodes 1 \
  --actor_num_gpus_per_node 2 \
  --vllm_num_engines 4 \
  --colocate_critic_reward \
  --colocate_actor_ref \
  --pretrain OpenRLHF/Llama-3-8b-sft-mixture \
  --reward_pretrain OpenRLHF/Llama-3-8b-rm-mixture \
  --save_path {DIR} \
  --micro_train_batch_size 8 \
  --train_batch_size 128 \
  --micro_rollout_batch_size 16 \
  --rollout_batch_size 1024 \
  --max_samples 100000 \
  --max_epochs 1 \
  --prompt_max_len 1024 \
  --generate_max_len 1024 \
  --zero_stage 3 \
  --bf16 \
  --actor_learning_rate 5e-7 \
  --critic_learning_rate 9e-6 \
  --init_kl_coef 0.01 \
  --prompt_data OpenRLHF/prompt-collection-v0.1 \
  --input_key context_messages \
  --apply_chat_template \
  --normalize_reward \
  --packing_samples \
  --adam_offload \
  --flash_attn \
  --gradient_checkpointing

**Comment by chanchimin** [comment]
I meet the same problem, how do you solve it, thank you

**Comment by ShoutaoGuo** [comment]
Maybe you should check your NCCL and IB configurations
